tests/spicy/types/sink/parse-error.spicy

```diff
@@ -0,0 +1,29 @@
+# @TEST-EXEC:  printf '1' | spicy-driver -d %INPUT >>output 2>&1
+# @TEST-EXEC:  printf 'A' | spicy-driver -d %INPUT >>output 2>&1
+# @TEST-EXEC:  btest-diff output
+#
+# @TEST-DOC: Check that parse errors of units connected to sinks aren't propagated back to the writer (#1076)
+
+module test;
+
+type S1 = unit {
+    x : /A/;
+    on %error { print "error in S1"; }
+    on %done { print self; }
+};
+
+type S2 = unit {
+    x : /./;
```

Comment:
Let's use a different name for this field so it is easier to distinguish it from `S1.x`.